### ISSUE TITLE
Memory limit for pipelinerunner (256Mi) is too low

### ISSUE THREAD

**Issue by funkypenguin** [jenkins-x/jx#4862]
### Summary 

On a serverless jenkins install with the kubernetes provider, quickstarts can be created, but after a small amount of interactions (1 or 2 PRs), pipelinerunner dies after 3 failed `git clone` attempts. Restarting the pipeline simply causes it to die again, but deleting the pod (_and allowing it to be recreated by the replicaset_) restores expected behaviour for 1-2 more PRs.

### Steps to reproduce the behavior

Install `golang-httpd` quickstart, using `jx create quickstart`. After the repo is created, create a PR. Observe the activity with `jx get activity`, and watch the logs of the pipelinerunner using `kubectl logs`. Submit PR and merge them (_I've never managed to do more than 2_) until the pipelinerunner starts to produce errors about `retrying after error:failed to init a new git repository in directory /tmp/`

### Expected behavior

pipelinerunner will continue to .. run pipelines :)

### Actual behavior

pipelinerunner dies with error about git cloning, and requires pod deletion to recover. Logs recorded are:

```
{"component":"pipelinerunner","level":"info","msg":"POST request","request":"POST / HTTP/1.1\r\nHost: pipelinerunner\r\nAccept-Encoding: gzip\r\nContent-Length: 284\r\nContent-Type: application/json\r\nUser-Agent: Go-http-client/1.1\r\n\r\n{\"labels\":{\"created-by-prow\":\"true\",\"prowJobName\":\"48f3ba19-af7f-11e9-b436-54b2031988f8\"},\"prowJobSpec\":{\"type\":\"postsubmit\",\"agent\":\"tekton\",\"refs\":{\"org\":\"fpjx\",\"repo\":\"environment-mistressglow-production\",\"base_ref\":\"master\",\"base_sha\":\"c7261674baed5d152cba45c5f51e1382ef6fb353\"}}}","time":"2019-07-26T08:28:03Z"}
{"level":"info","msg":"Generated Git credentials file /home/jenkins/git/credentials","time":"2019-07-26T08:28:03Z"}
{"branch":"master","component":"pipelinerunner","context":"","level":"info","meta":false,"msg":"triggering pipeline","revision":"c7261674baed5d152cba45c5f51e1382ef6fb353","sourceURL":"https://github.com/fpjx/environment-mistressglow-production.git","time":"2019-07-26T08:28:03Z"}
{"level":"info","msg":"Deleting and cloning the Jenkins X versions repo","time":"2019-07-26T08:28:03Z"}
{"level":"info","msg":"Cloning the Jenkins X versions repo https://github.com/jenkins-x/jenkins-x-versions.git with ref refs/heads/master to /root/.jx/jenkins-x-versions","time":"2019-07-26T08:28:03Z"}
{"level":"info","msg":"shallow cloning repository https://github.com/fpjx/environment-mistressglow-production.git to temp dir /tmp/git679694225","time":"2019-07-26T08:28:08Z"}
{"level":"warning","msg":"\nretrying after error:failed to init a new git repository in directory /tmp/git679694225: git output: : failed to run 'git init' command in directory '/tmp/git679694225', output: ''\n","time":"2019-07-26T08:28:12Z"}
{"level":"info","msg":"shallow cloning repository https://github.com/fpjx/environment-mistressglow-production.git to temp dir /tmp/git852096444","time":"2019-07-26T08:28:12Z"}
{"level":"warning","msg":"\nretrying after error:failed to init a new git repository in directory /tmp/git852096444: git output: : failed to run 'git init' command in directory '/tmp/git852096444', output: ''\n","time":"2019-07-26T08:28:14Z"}
{"level":"info","msg":"shallow cloning repository https://github.com/fpjx/environment-mistressglow-production.git to temp dir /tmp/git581294571","time":"2019-07-26T08:28:14Z"}
{"level":"fatal","msg":"failed to clone three times it's likely things wont recover so lets kill the process; after 3 attempts, last error: failed to init a new git repository in directory /tmp/git581294571: git output: : failed to run 'git init' command in directory '/tmp/git581294571', output: ''","time":"2019-07-26T08:28:14Z"}

```

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.517
jenkins x platform 2.0.880
Kubernetes cluster v1.14.2
kubectl            v1.15.0
helm client        Client: v2.14.1+g5270352
git                git version 2.17.1
Operating System   Ubuntu 18.04.2 LTS
```

### Jenkins type

- [X] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

Kubernetes 1.14 on-prem, created using kubeadm

### Operating system / Environment

Ubuntu 1804

### JX install parameters

My jx install parameters were as follows:

```
jx install --provider=kubernetes \
--default-admin-username=admin \
--default-admin-password=admin \
--git-username funkypenguin \
--git-api-token $MYTOKEN \
--skip-ingress \
--external-ip=202.137.244.245 \
--domain=elpenguino.net \
--tekton \
--no-tiller \
--prow \
--docker-registry=registry-internal.elpenguino.net \
--docker-registry-org=library \
--ingress-class=nginx-external \
--external-dns=false
```

**Comment by funkypenguin** [comment]
Duplicate of https://github.com/jenkins-x/jx/issues/4593, it turns out (thanks @hferentschik )

**Comment by funkypenguin** [comment]
So the fragility of the pod is related to #4593, but it turns out that the reason for this error is simply that 256Mi (the limit assigned to the pipelinerunner) is too low for `jx controller pipelinerunner`.

I was able to reproduce this error after 1 or 2 PRs - I'd then exec into the container, and attempt to run `git clone` (any repo), and even `yum update`. In all cases, commands would exit with error 143 (OOM kill).

dmesg reported (after I tried `yum update`) the following:

```
[45490.361101] Memory cgroup out of memory: Kill process 22609 (yum) score 1061 or sacrifice child
[45490.362991] Killed process 22609 (yum) total-vm:218924kB, anon-rss:11456kB, file-rss:6724kB, shmem-rss:0kB
[root@pipelinerunner-54d859d7d4-hlcz7 jenkins]#
```

I manually edited the deployment for pipelinerunner, and increased the memory limit from 256Mi to 512Mi (as well as the request, since clearly we exceed 256Mi), and have now processed 8+ PRs without incident.

I'd recommend that the memory limit on the pipeline runner be permanently increased to meet maxium `jx controller pipelinerunner` requirements.

D

**Comment by hferentschik** [comment]
@funkypenguin, thanks a lot for all the debugging related to this bug. Much appreciated 👍

**Comment by hferentschik** [comment]
I created https://github.com/jenkins-x-charts/prow/pull/976 to increase the memory limit in the Prow chart.

**Comment by hferentschik** [comment]
Update to Prow charts merged via jenkins-x-charts/prow#976. To upgrade current prod env https://github.com/jenkins-x/environment-tekton-mole-dev/pull/226.
